src/System.Private.CoreLib/shared/System/Number.Formatting.cs

```diff
@@ -1094,18 +1094,30 @@ private static unsafe bool TryInt32ToHexStr(int value, char hexBase, int digits,
         [MethodImpl(MethodImplOptions.AggressiveInlining)] // called from only one location
         private static unsafe void UInt32ToNumber(uint value, ref NumberBuffer number)
         {
-            number.Precision = UInt32Precision;
-            number.Sign = false;
+            number.DigitsCount = UInt32Precision;
+            number.IsNegative = false;
 
-            char* buffer = number.GetDigitsPointer();
-            char* p = UInt32ToDecChars(buffer + UInt32Precision, value, 0);
+            byte* buffer = number.GetDigitsPointer();
+            byte* p = UInt32ToDecChars(buffer + UInt32Precision, value, 0);
             int i = (int)(buffer + UInt32Precision - p);
             number.Scale = i;
 
-            char* dst = number.GetDigitsPointer();
+            byte* dst = number.GetDigitsPointer();
             while (--i >= 0)
                 *dst++ = *p++;
-            *dst = '\0';
+            *dst = (byte)('\0');
+        }
+
+        internal static unsafe byte* UInt32ToDecChars(byte* bufferEnd, uint value, int digits)
```

Comment:
Is the `char*` variant still used after this is introduced?

Reply:
Yes, in the formatter. I think we could probably update this to be shareable though.